src/cdk/schematics/ng-update/devkit-file-system.ts

```diff
@@ -77,4 +69,15 @@ export class DevkitFileSystem extends FileSystem {
     const {subdirs: directories, subfiles: files} = this._tree.getDir(dirPath);
     return {directories, files};
   }
+
+  private _isExistingDirectory(dirPath: Path) {
+    if (dirPath === NormalizedRoot) {
```

Comment:
Is this supposed to be `instanceof NormalizedRoot` or is there something passing in the actual class?

Reply:
This is to check if the given directory is root (NormalizedRoot = `/`). If it is root, we can just assume it exists.
If it is not root, the normal parent/child check happens.
Or did I misunderstand your question?

Reply:
The part I'm confused by is that we're looking for a specific class here (`NormalizedRoot`) and I wasn't sure whether that's intentional. So you're saying that `NormalizedRoot` is a special value passed in only for the root directory?

Reply:
Ah, I see
Yes, as in Path and NormalizedRoot are simply typed strings. Due to this the `===` comparison works, which I believe is intended by the Angular CLI Devkit.
https://github.com/angular/angular-cli/blob/master/packages/angular_devkit/core/src/virtual-fs/path.ts#L49